xinference/model/llm/ggml/tests/test_ctransformers.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2022-2023 XProbe Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import random
+import string
+
+import pytest
+
+from .....client import Client, GenerateModelHandle
+from ....llm import GgmlLLMSpecV1, LLMFamilyV1
+from ..ctransformers import CtransformersModel
+
+
+class MockCtransformersModel(CtransformersModel):
+    def load(self):
+        pass
+
+
+mock_model_spec = GgmlLLMSpecV1(
+    model_format="ggmlv3",
+    model_size_in_billions=6,
+    quantizations=["q2_k", "q4_0"],
+    model_id="test_id",
+    model_file_name_template="TestModel.{quantization}.ggmlv3.bin",
+)
+
+test_model_spec = """{
+   "version":1,
+   "model_name":"TestModel",
+   "model_lang":[
+      "en"
+   ],
+   "model_ability":[
+      "embed", "generate"
+   ],
+   "model_specs":[
+      {
+         "model_format":"ggmlv3",
+         "model_size_in_billions":6,
+         "quantizations": ["q2_k", "q4_0"],
+         "model_id":"test_id",
+         "model_file_name_template":"TestModel.{quantization}.ggmlv3.bin"
+      },
+      {
+         "model_format":"pytorch",
+         "model_size_in_billions":3,
+         "quantizations": ["int8", "int4", "none"],
+         "model_id":"example/TestModel"
+      }
+   ],
+   "prompt_style": null
+}"""
+
+mock_model_family = LLMFamilyV1.parse_raw(test_model_spec)
+
+
+@pytest.mark.parametrize(
+    "model_spec, model_family", [(mock_model_spec, mock_model_family)]
+)
+def test_ctransformer_init(model_spec, model_family):
+    from ctransformers import AutoConfig
+
+    quantization = "q4_0"
+    uid = "".join(random.choice(string.digits) for i in range(15))
+    path = "".join(
+        random.choice(string.ascii_letters + string.punctuation) for i in range(100)
+    )
+    model = MockCtransformersModel(
```

Comment:
Don't need a mock class since the method `load` is not invoked.

Reply:
changed to the original Ctransformers class